hack/dockerized

```diff
@@ -12,7 +12,20 @@ fi
 
 fail_if_cri_bin_missing
 
-KUBEVIRT_BUILDER_IMAGE=${KUBEVIRT_BUILDER_IMAGE:-"quay.io/kubevirt/builder:2405211226-1c571a3c4"}
+kubevirt_builder_version="2405211226-1c571a3c4"
+default_kubevirt_builder_image="quay.io/kubevirt/builder:${kubevirt_builder_version}"
+kubevirt_cross_compile_image="quay.io/kubevirt/builder-cross-compile:${kubevirt_builder_version}"
+
+KUBEVIRT_BUILDER_IMAGE=${KUBEVIRT_BUILDER_IMAGE:-"${default_kubevirt_builder_image}"}
+
+if [ "${KUBEVIRT_BUILDER_IMAGE}" == "${default_kubevirt_builder_image}" ] && [ -n "${BUILD_ARCH}" ]; then
+    for arch in ${BUILD_ARCH//,/ }; do
+        arch=$(format_archname "${arch}")
+        if [[ ${arch} =~ ^cross-* ]]; then
+            KUBEVIRT_BUILDER_IMAGE="${kubevirt_cross_compile_image}"
```

Comment:
IIUC this wouldn't allow you to override both the native and cross builder image at the same time, which I think is something that would make sense.

So we should have `KUBEVIRT_CROSS_BUILDER_IMAGE` too.

Reply:
My idea was to allow `KUBEVIRT_BUILDER_IMAGE` to always override the image used, as my reasoning is that if you are using it, you know what you are doing. That's why i did not add a second variable for cross-compiling, but rather check if `KUBEVIRT_BUILDER_IMAGE` is set by user one line above:
```
if [ "${KUBEVIRT_BUILDER_IMAGE}" == "${default_kubevirt_builder_image}" ] && [ -n "${BUILD_ARCH}" ]; then
    for arch in ${BUILD_ARCH//,/ }; do
```

Reply:
Yeah, okay, I'm not sure whether there are potential pitfalls that I haven't considered but in general this sounds like it could work. Might be worth having a short comment explaining the logic though.

Reply:
I have added a comment before the first if

Reply:
Looks good, thanks!